modin/tests/pandas/dataframe/test_join_sort.py

```diff
@@ -296,7 +298,9 @@ def test_merge(test_data, test_data2):
                 right_on="key",
                 sort=sorts[j],
             )
-            sort_if_range_partitioning(modin_result, pandas_result)
+            sort_if_range_partitioning(
+                modin_result, pandas_result, force=StorageFormat.get() == "Hdk"
```

Comment:
Looks like `HDK` doesn't preserve the same order as pandas. I don’t create an issue for this, since the engine is deprecated.

Reply:
Just to note, Modin on Ray/Dask/MPI may also have mismatched order - https://github.com/modin-project/modin/issues/2246.